### PR TITLE
Move transaction queue code into separate package

### PR DESCRIPTION
**Pull request by divan**
This PR is part of refactoring packages process and moves all transaction queue related code into separate package.

### CODE REVIEW

File: geth/txqueue/utils.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/status-im/status-go/geth/common"
+	"github.com/status-im/status-go/geth/signal"
+)
+
+var ErrTxQueueRunFailure = errors.New("error running transaction queue")
```
Comment (adambabik) [inline geth/txqueue/utils.go]: Hm, it looks like this is not used anywhere except for a `fmt.Errorf()` below.

File: geth/txqueue/utils.go [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// HaltOnPanic recovers from panic, logs issue, sends upward notification, and exits
+func HaltOnPanic() {
```
Comment (adambabik) [inline geth/txqueue/utils.go]: So this code is duplicated in `node/utils.go`?

Comment (divan) [reply geth/txqueue/utils.go]: Yes. I'm not sure it's even a right thing to have here, but for the sake of keeping the same functionality, I just copied code. Not sure it's worth moving into separate package, I think we'll revisit the queue halting logic soon as well (I don't like that tqQueue send signals here, it's wrong place to handle go-react communication)

File: geth/txqueue/utils.go (hunk 1 above)
Comment (divan) [reply geth/txqueue/utils.go]: Correct. Idea was to introduce as little changes as possible in this PR. Just extracting code into separate package.

File: geth/txqueue/utils.go (hunk 2 above)
Comment (adambabik) [reply geth/txqueue/utils.go]: Makes sense. I just wanted to draw attention so that all of us are on the same page.

**Comment by divan** [comment]
> Where is HaltOnPanic() used? Btw, its comment misses the dot. 

@themue can you please comment in the code review UI  next time? So the comment is under the code itself - it will make possible replying directly and keeping conversation in thread. This way discussion can only be done via higher-level comments, which is not very convinient.

**Pull request merged**